keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/KeelRepository.kt

```diff
@@ -231,4 +270,108 @@ interface KeelRepository {
   ): ArtifactSummaryInEnvironment?
 
   // END ArtifactRepository methods
+
+  fun getResourceEventHistory(id: String, limit: Int = ResourceRepository.DEFAULT_MAX_EVENTS): List<PersistentEvent> {
```

Comment:
Functions dealing with resource event history and status now live here, which is the closest place we have at the moment to a service object for these kinds of operations.